hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/test/java/org/apache/hadoop/yarn/server/resourcemanager/monitor/capacity/TestProportionalCapacityPreemptionPolicy.java

```diff
@@ -1083,46 +1083,67 @@ public void testRefreshPreemptionProperties() throws Exception {
   }
 
   @Test
-  public void testLeafQueueNameExtraction() throws Exception {
-    ProportionalCapacityPreemptionPolicy policy =
-        buildPolicy(Q_DATA_FOR_IGNORE);
+  public void testLeafQueueNameExtractionWithFlexibleAQC() throws Exception {
+    runLeafQueueNameExtractionTest(true);
+  }
+
+  @Test
+  public void testLeafQueueNameExtractionWithLegacyAQC() throws Exception {
+    runLeafQueueNameExtractionTest(false);
+  }
+
+  private void runLeafQueueNameExtractionTest(boolean isFlexible) throws Exception {
```

Comment:
NIT: maybe this runLeafQueueNameExtractionTest can be refactored to remove the flag parameter.

https://martinfowler.com/bliki/FlagArgument.html